### PR TITLE
fix: add fail index to streaming sequence

### PR DESCRIPTION
**Pull request by leahecole**
Gal was working on this before he moved to another team and I am opening this to make sure I tie up loose ends by finishing his work. I added a test and then I regenerated the support files because of the proto change. It seemed to regenerate more than just my service, but I removed anything unrelated to sequence. If I should re-include those, let me know. cc @galz10

### CODE REVIEW

**Review by noahdietz — APPROVED** [review]
I don't think you need to include the generated code changes in this PR. They should be automatically included in CI for testing, then contributed back via an automatic PR upon merge.

File: schema/google/showcase/v1beta1/sequence.proto [hunk 1]
```diff
@@ -217,14 +217,19 @@ message AttemptSequenceRequest {
...
 
 message AttemptStreamingSequenceRequest {
   string name = 1 [
     (google.api.resource_reference).type = "showcase.googleapis.com/StreamingSequence",
     (google.api.field_behavior) = REQUIRED
   ];
-
+  
+  int32 fail_index = 2 [
+    (google.api.resource_reference).type = "showcase.googleapis.com/Sequence",
```
Comment (blakeli0) [inline schema/google/showcase/v1beta1/sequence.proto:230]: @noahdietz Is this is a valid resource reference? I think the field reference another resource should be of String type per [AIP-122](https://google.aip.dev/122), also the [Sequence](https://github.com/googleapis/gapic-showcase/blob/fe2fe253b030604605ea926b7e4664585669969d/schema/google/showcase/v1beta1/sequence.proto#L92-L95) resource is of pattern `sequences/{sequence}` which seems do not fit here.

File: schema/google/showcase/v1beta1/sequence.proto [hunk 2]
```diff
@@ -217,14 +217,19 @@ message AttemptSequenceRequest {
...
 }
 
 message AttemptStreamingSequenceRequest {
   string name = 1 [
     (google.api.resource_reference).type = "showcase.googleapis.com/StreamingSequence",
     (google.api.field_behavior) = REQUIRED
   ];
-
+  
+  int32 fail_index = 2 [
```
Comment (blakeli0) [inline schema/google/showcase/v1beta1/sequence.proto:229]: Can we add some comments to explain what this field is used for? Also based on the server logic, this looks more like a `last_failed_index` or `restart_index`?

File: schema/google/showcase/v1beta1/sequence.proto (hunk 1 above)
Comment (noahdietz) [reply schema/google/showcase/v1beta1/sequence.proto:230]: Oh yeah definitely not.

Comment (noahdietz) [inline schema/google/showcase/v1beta1/sequence.proto:230]:
```suggestion
  int32 fail_index = 2 [
```

File: schema/google/showcase/v1beta1/sequence.proto (hunk 2 above)
Comment (leahecole) [reply schema/google/showcase/v1beta1/sequence.proto:229]: good call, I like `last_failed_index`

File: schema/google/showcase/v1beta1/sequence.proto (hunk 1 above)
Comment (leahecole) [reply schema/google/showcase/v1beta1/sequence.proto]: good catch! I accepted Noah's suggestion which iiuc just removed what used to be line 30 with this reference type. I added a clarifying comment, good call @blakeli0. I want to make sure I'm interpreting these comments and the code suggestion from @noahdietz correctly though - do I not need to include a resource type at all? Or, should I be linking to something like https://github.com/googleapis/gapic-showcase/blob/3188c8a6089597d937ed6ab5b529716ab92d5ff3/schema/google/showcase/v1beta1/sequence.proto#L233?

Comment (noahdietz) [reply schema/google/showcase/v1beta1/sequence.proto]: Nope, not necessary. The resource here is the Sequence being attempted. The `content` is just data.

**Review by blakeli0 — APPROVED** [review]

**Pull request merged**